app/controllers/application_controller.rb

```diff
@@ -1,2 +1,61 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  include CurrentUser
+  include JsonResolver
+
+  protect_from_forgery unless: -> { request.format.json? }
+
+  rescue_from Exception do |e|
+    error(e)
+  end
+
+  # 400
+  rescue_from Exceptions::BadRequest, ActiveRecord::RecordInvalid, ActionController::ParameterMissing, ArgumentError do |e|
+    exception_handler e, :bad_request
+  end
+
+  # 401
+  rescue_from Exceptions::Unauthorized do |e|
+    exception_handler e, :unauthorized
+  end
+
+  # 403
+  rescue_from Exceptions::Forbidden do |e|
+    exception_handler e, :forbidden
+  end
+
+  # 404
+  rescue_from Exceptions::NotFound, ActiveRecord::RecordNotFound do |e|
+    exception_handler e, :not_found
+  end
+
+  # 409
+  rescue_from Exceptions::Conflict, ActiveRecord::RecordNotDestroyed do |e|
+    exception_handler e, :conflict
+  end
+
+  private
+
+  def authenticate_request
+    render json: { message: '로그인이 필요합니다' }, status: Exceptions::Unauthorized.http_status if current_user.blank?
```

Comment:
`Exceptions::Unauthorized` 클래스도 `rescue_from`으로 정의가 되어 있으니 아래와 같이 바로 `raise` 시키면 될 것 같네요.

```suggestion
    raise Exceptions::Unauthorized, '로그인이 필요합니다' if current_user.blank?
```